src/engine/memorytable.cpp

```diff
@@ -338,7 +338,7 @@ quint32 MemoryTablePrivate::allocate(quint32 size, TableMetadata *table, bool in
 
     // Align the allocation so that the header is directly accessible
     quint32 allocationSize = static_cast<quint32>(requiredSpace(size));
```

Comment:
Maybe this should return a size_t, to avoid useless static_cast, and enforce 64 bits compatibility